pd/migrations/20211216025603_init.sql

```diff
@@ -113,30 +106,59 @@ CREATE TABLE IF NOT EXISTS delegation_changes (
 CREATE INDEX ON delegation_changes (epoch);
 CREATE INDEX ON delegation_changes (validator_identity_key);
 
--- Set of quarantined notes: once the epoch is large enough, all old-enough quarantined notes should
--- be inserted into the NCT and the notes table (and removed from this table)
+-- Nullifiers, indexed by height
+CREATE TABLE IF NOT EXISTS nullifiers (
+    nullifier bytea PRIMARY KEY,
+    height bigint NOT NULL REFERENCES blocks (height),
+
+    -- if this nullifier was quarantined at any point in the past:
+    unbonding_height bigint, -- height at which to make the spend permanent
+    reverted_height bigint REFERENCES blocks (height), -- height at which the revert was made, if it was reverted
+    validator_identity_key bytea REFERENCES validators (identity_key),
```

Comment:
I don't think it's a good idea to mix this data in with the nullifier set.  Checking whether something is in the nullifier set should be a simple set membership query, we should not have to handle other data, or deal with whether or not these fields are present or absent.